### PR TITLE
Implement the "remove friend" button

### PR DESCRIPTION
**Pull request by MathisHage**
This pull request introduces a new feature to delete friends from the user's friend list and includes several related updates in the view model, user interface, and tests.

### New Feature: Delete Friend

* **ViewModel Update:**
  - Added `deleteFriend` method in `UserProfileViewModel` to handle the removal of friends from the user's friend list.

* **UI Enhancements:**
  - Added `DeleteFriendButton` composable function in `FriendsScreen.kt` to provide a UI element for deleting friends.
  - Updated `FriendItem` composable function to include the `DeleteFriendButton`.
  - Modified `ViewFriendsScreen` to pass `userProfileViewModel` to `FriendItem`.

* **Imports and Dependencies:**
  - Added necessary imports for the delete icon and color in `FriendsScreen.kt`.
 
* **Testing:**
  - Added unit tests in `UserProfileViewModelTest` to verify the functionality of `deleteFriend` and `addFriend` methods.

### CODE REVIEW

File: app/src/main/java/com/github/se/orator/model/profile/UserProfileViewModel.kt [hunk 1]
```diff
@@ -263,4 +263,35 @@ class UserProfileViewModel(internal val repository: UserProfileRepository) : Vie
...
+        userProfile_.value = updatedProfile
+
+        // Updates the local state with the new friends list
+        friendsProfiles_.value = updatedFriendsList
+      } else {
+        Log.d("UserProfileViewModel", "${friend.name} cannot be deleted: not in the friends list !")
+      }
+    } else {
+      Log.e("UserProfileViewModel", "Failed to remove a friend: current user profile is null.")
+    }
```
Comment (charbelraffoul) [inline app/src/main/java/com/github/se/orator/model/profile/UserProfileViewModel.kt:295]: It’s good that you added checks for both currentUserProfile and friend presence in the list before attempting removal. This ensures the method doesn't fail silently or perform unnecessary operations.

File: app/src/main/java/com/github/se/orator/model/profile/UserProfileViewModel.kt [hunk 2]
```diff
@@ -263,4 +263,35 @@ class UserProfileViewModel(internal val repository: UserProfileRepository) : Vie
 
     return profile == null || profile.name.isBlank()
   }
+
+  /**
+   * Deletes a friend from the current user's list of friends.
+   *
+   * @param friend The `UserProfile` of the friend to be deleted.
+   */
+  fun deleteFriend(friend: UserProfile) {
```
Comment (charbelraffoul) [inline app/src/main/java/com/github/se/orator/model/profile/UserProfileViewModel.kt:272]: Suggested Improvement:
Currently, the deleteFriend method doesn’t return a result. If this method is intended for testing or error handling, consider returning a Boolean or status code indicating success or failure. This can help in tracking issues in larger codebases.

File: app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt [hunk 3]
```diff
@@ -212,6 +216,9 @@ fun FriendItem(friend: UserProfile) {
                 overflow = TextOverflow.Ellipsis,
                 modifier = Modifier.testTag("friendBio#${friend.uid}"))
           }
+          Spacer(modifier = Modifier.weight(1f)) // Pushes the delete button to the right
```
Comment (charbelraffoul) [inline app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt:219]: The DeleteFriendButton composable is integrated well within the FriendItem. It’s smartly positioned to the right, making it accessible without crowding the UI.

File: app/src/test/java/com/github/se/orator/model/profile/UserProfileViewModelTest.kt [hunk 4]
```diff
@@ -188,4 +188,68 @@ class UserProfileViewModelTest {
     val selectedFriend = viewModel.selectedFriend.first()
     Assert.assertEquals(friendProfile, selectedFriend)
   }
+
```
Comment (charbelraffoul) [inline app/src/test/java/com/github/se/orator/model/profile/UserProfileViewModelTest.kt:191]: The test cases check friend list size before and after modification, which is great for verifying list integrity. Consider adding some documentation and comment for readers to understand better the test cases, if we ever need to modify them.

**Review by charbelraffoul — APPROVED** [review]
Thank you for your work on this PR! This feature is well thought-out, with clear improvements in the ViewModel and UI components. The added tests also do a great job of covering key scenarios, which will help maintain code quality. Here are a few suggestions to consider as you finalize the PR:

Error Handling in deleteFriend: Consider returning a Boolean or status code from the deleteFriend method to indicate success or failure, which can improve error tracking in larger codebases.

Test Edge Cases: It would be helpful to add edge case tests, such as attempting to delete a friend not in the list or handling a null currentUserProfile. This will strengthen the coverage and ensure the ViewModel can gracefully handle unexpected scenarios.

Test Naming Consistency: For easier understanding, consider using more descriptive names for your test functions, specifying both conditions and expected outcomes.

Thank you again for your thorough work on this feature! I’m approving the PR, and these suggestions can be addressed at a later time. Let me know if you have questions or need any more details on these points.
